__tests__/test-data.test.js

```diff
@@ -0,0 +1,166 @@
+        .get("/api/topic")
+        // wrong end point name
+        .expect(404)
+    );
+  });
+});
+
+
+
+
+
+
+
+
+
+describe("get /api", () => {
+  test("200,returns an object ", () => {
+    return request(app)
+      .get("/api")
+      .expect(200)
+      .then((body) => {
+        const endPointData = body.body.endPointData;
+        expect(Array.isArray(endPointData)).toBe(false);
+        expect(typeof endPointData).toBe("object");
+      });
+  });
+  test("200, check if the key in each object matches the api path and contains information  ", () => {
+    return request(app)
+      .get("/api")
+      .expect(200)
+      .then((body) => {
+        const endPointData = body.body.endPointData;
+        const arraKey = Object.keys(endPointData);
+        expect(arraKey).toEqual([
+          "GET /api",
+          "GET /api/topics",
+          "GET /api/articles",
+        ]);
+      });
+  });
+  test("200, check if each nested obj contains a description with a information inisde it  ", () => {
+    return request(app)
+      .get("/api")
+      .expect(200)
+      .then((body) => {
+        const endPointData = body.body.endPointData;
+        const endPointKeys = Object.keys(endPointData);
+        const everykeyHasDescriptionInfo = endPointKeys.every((key) => {
+          if (endPointData[key].description !== null) {
+            return true;
+          }
+        });
+        for (const key of endPointKeys) {
+        }
+        expect(everykeyHasDescriptionInfo).toBe(true);
+      });
+  });
+  test("200, check if each nested obj, other than GET /api, contains a queries key with a information inisde it  ", () => {
+    return request(app)
+      .get("/api")
+      .expect(200)
+      .then((body) => {
+        const endPointData = body.body.endPointData;
+        const endPointKeys = Object.keys(endPointData);
+        const firstElementRemodedArray = endPointKeys.slice(1);
+        const everykeyHasQueryInfo = firstElementRemodedArray.every((key) => {
+          return endPointData[key].queries !== undefined;
+        });
+        expect(everykeyHasQueryInfo).toBe(true);
+      });
+  });
+  test("200, check if response body has a specific struvture it wants you to follow ", () => {
+    return request(app)
+      .get("/api")
+      .expect(200)
+      .then((body) => {
+        const endPointData = body.body.endPointData;
+        const exampleResponseKeys = Object.keys(endPointData);
+        const firstElementRemodedArray = exampleResponseKeys.slice(1);
+        let result = false;
+        for (const key of firstElementRemodedArray) {
+          const obj = endPointData[key].exampleResponse;
+          const keyArr = Object.keys(obj);
+          let objData = obj[keyArr[0]];
+          if (Object.keys(objData).length > 0) {
+            result = true;
+          }
+        }
+
+        expect(result).toBe(true);
+      });
+  });
+  test("200, check if response body gives an example of what a response looks like  ", () => {
```

Comment:
this test is really good but doing it like this could become unmanageable, because the endpoints.json is a JSON file you can just require it and check that the response equals the same as the file